tidb-lightning/tidb-lightning-error-resolution.md

```diff
@@ -0,0 +1,309 @@
+| ------------ | ------ | ---- | ---------- | ----------------------------------------------------------------------------------------------------------------------------------- |
+| task_id      | ✓      | ✓    | ✓        | 生成此错误的 TiDB Lightning 任务 ID                                            |
+| create_table | ✓      | ✓    | ✓        | 记录错误的时间                                                                   |
+| table_name   | ✓      | ✓    | ✓        | 包含错误的表的名称，格式为 ``'`db`.`tbl`'``                                                                |
+| path         | ✓      | ✓    |          | 包含错误文件的路径                                                       |
+| offset       | ✓      | ✓    |          | 文件中发现错误的字节位置                                         |
+| error        | ✓      | ✓    |          | 错误信息                                                                                 |
+| context      | ✓      |      |          | 围绕错误的文本                                                             |
+| index_name   |        |      | ✓        | 冲突中唯一键的名称。主键冲突为 `'PRIMARY'`                    |
+| key_data     |        |      | ✓        | 导致错误的行的格式化键句柄。该内容仅供人参考，机器不可读 |
+| row_data     |        | ✓    | ✓        | 导致错误的格式化行数据。该内容仅供人参考，机器不可读          |
+| raw_key      |        |      | ✓        | 冲突的 KV 对的键                                                           |
+| raw_value    |        |      | ✓        | 冲突的 KV 对的值                                                            |
+| raw_handle   |        |      | ✓        | 冲突行的行句柄                                                         |
+| raw_row      |        |      | ✓        | 冲突行的编码值                                                       |
+
+> **注意：**
+>
+> 错误报告记录的是文件偏移量，不是行号或列号，因为行号或列号的获取效率很低。你可以使用下列命令在字节位实现快速跳转（以 183 为例）：
+>
+> * shell, 输出前面几行——
+>
+>     ```sh
+>     head -c 183 file.csv | tail
+>     ```
+>
+> * shell, 输出后面几行——
+>
+>     ```sh
+>     tail -c +183 file.csv | head
+>     ```
+>
+> * vim — `:goto 183` 或 `183go`
+
+## 示例
+
+在该示例中，我们准备了一个包含一些已知错误的数据源。以下是处理这些错误的具体步骤：
+
+1. 准备数据库和表结构：
+
+    {{< copyable "shell-regular" >}}
+
+    ```sh
+    mkdir example && cd example
+    echo 'CREATE SCHEMA example;' > example-schema-create.sql
+    echo 'CREATE TABLE t(a TINYINT PRIMARY KEY, b VARCHAR(12) NOT NULL UNIQUE);' > example.t-schema.sql
+    ```
+
+2. 准备数据：
+
+    {{< copyable "shell-regular" >}}
+
+    ```sh
+    cat <<EOF > example.t.1.sql
+        INSERT INTO t (a, b) VALUES
+        (0, NULL),              -- 列不为空
+        (1, 'one'),
+        (2, 'two'),
+        (40, 'forty'),          -- 与下面的 `40` 冲突
+        (54, 'fifty-four'),     -- 与下面的 `54` 冲突
+        (77, 'seventy-seven'),  -- 字符串长度超过 12 个字符
+        (600, 'six hundred'),   -- 数字超出了 TINYINT 数据类型支持的范围
+        (40, 'fourty'),         -- 与上面的 `40` 冲突
+        (42, 'fifty-four');     -- 与上面的 `54` 冲突
```

Comment:
```suggestion
        (42, 'fifty-four');     -- 与上面的 `54` 冲突
```
done~ PTAL. And I wonder if the number "42" should be "54"? @kennytm

Reply:
@Liuxiaozhen12  42 is correct.

Reply:
```suggestion
        (42, 'fifty-four');     -- 与上面的 `'fifty-four'` 冲突
```